http-server/src/main/java/com/facebook/airlift/http/server/SecureServlet.java

```diff
@@ -0,0 +1,121 @@
+import java.io.InputStream;
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.io.ByteStreams.copy;
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SecureServlet
+        extends HttpServlet
+{
+    private final Servlet delegate;
+    private final Authorizer authorizer;
+    private final Optional<Set<String>> allowedRoles;
+    private final HttpServerConfig httpServerConfig;
+
+    public SecureServlet(Servlet delegate,
+            Authorizer authorizer,
+            Optional<Set<String>> allowedRoles,
+            HttpServerConfig httpServerConfig)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+        this.authorizer = requireNonNull(authorizer, "authorizer is null");
+        this.allowedRoles = requireNonNull(allowedRoles, "allowedRoles is null");
+        this.httpServerConfig = requireNonNull(httpServerConfig, "httpServerConfig is null");
+    }
+
+    @Override
+    public void service(ServletRequest req, ServletResponse res)
+            throws ServletException, IOException
+    {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) res;
+
+        Principal principal = request.getUserPrincipal();
+        if (principal == null) {
+            skipRequestBody(request);
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "Request principal is missing.");
+            return;
+        }
+
+        Optional<Set<String>> allowedRoles = this.allowedRoles;
+        if (isEndpointUnmarked(allowedRoles)) {
+            switch (httpServerConfig.getDefaultAuthorizationPolicy()) {
+                case ALLOW:
+                    delegate.service(req, res);
+                    return;
+                case DENY:
+                    skipRequestBody(request);
+                    response.sendError(
+                            HttpServletResponse.SC_FORBIDDEN,
+                            format("Principal %s is not allowed to access the resource. Reason: denied by default policy",
+                                    principal.getName()));
+                    return;
+                case DEFAULT_ROLES:
+                    allowedRoles = Optional.of(httpServerConfig.getDefaultAllowedRoles());
+                    break;
+                default:
+            }
+        }
+
+        AuthorizationResult result = authorizer.authorize(principal, allowedRoles.get());
+        if (!result.isAllowed()) {
+            skipRequestBody(request);
+            response.sendError(
+                    HttpServletResponse.SC_FORBIDDEN,
+                    format("Principal %s is not allowed to access the resource. Reason: %s",
+                            principal.getName(),
+                            result.getReason()));
+            return;
```

Comment:
lets encapsulate this logic in a function and use it in all the places where you are sending a SC_FORBIDDEN